src/solver/infeasible-problem-analysis/constraint.cpp

```diff
@@ -39,20 +41,45 @@ double Constraint::getSlackValue() const
 {
     return mSlackValue;
 }
-std::vector<std::string> split(const std::string& s, char delimiter)
+
+class StringIsNotWellFormated : public std::runtime_error
 {
-   std::vector<std::string> tokens;
-   std::string token;
-   std::istringstream tokenStream(s);
-   while (std::getline(tokenStream, token, delimiter))
-   {
-      tokens.push_back(token);
-   }
-   return tokens;
-}
+public:
+    StringIsNotWellFormated(const std::string& error_message) : std::runtime_error(error_message)
+    {
+    }
+};
 
-std::string StringBetweenAngleBrackets(const std::string& str){
-  return split(split(str, '<')[1], '>')[0];
+std::string StringBetweenAngleBrackets(const std::string& str)
+{
+    const auto& str_begin_iterator = str.begin();
+    const auto& str_end_iterator = str.end();
```

Comment:
`str.begin` is actually shorter than `str_begin_iterator`, consider removing this alias reference